apps/papra-server/src/modules/documents/documents.routes.ts

```diff
@@ -422,3 +425,45 @@ function setupDeleteAllTrashDocumentsRoute({ app, config, db }: RouteDefinitionC
     },
   );
 }
+
+function setupUpdateDocumentContentRoute({ app, db }: RouteDefinitionContext) {
+  app.patch(
+    '/api/organizations/:organizationId/documents/:documentId',
+    validateParams(z.object({
+      organizationId: z.string().regex(organizationIdRegex),
+      documentId: z.string(),
+    })),
+    validateJsonBody(z.object({
+      content: z.string(),
+    })),
+    async (context) => {
+      const { userId } = getUser({ context });
+      const { organizationId, documentId } = context.req.valid('param');
+      const { content } = context.req.valid('json');
+
+      const documentsRepository = createDocumentsRepository({ db });
+      const organizationsRepository = createOrganizationsRepository({ db });
+
+      await ensureUserIsInOrganization({ userId, organizationId, organizationsRepository });
+
+      await db
+        .update(documentsTable)
+        .set({
+          content,
+          updatedAt: new Date(),
+        })
+        .where(
+          and(
+            eq(documentsTable.id, documentId),
+            eq(documentsTable.organizationId, organizationId),
+          ),
+        );
+
+      const { document } = await getDocumentOrThrow({ documentId, organizationId, documentsRepository });
```

Comment:
No need to refetch the document, it can be returned after updating to avoid a db call (still to do in repositories)

```suggestion
      const [document] = await db
        .update(documentsTable)
        .set({
          content,
          updatedAt: new Date(),
        })
        .where(
          and(
            eq(documentsTable.id, documentId),
            eq(documentsTable.organizationId, organizationId),
          ),
        )
        .returning()
```